src/basic/socket-util.c

```diff
@@ -1454,6 +1454,25 @@ int socket_bind_to_ifindex(int fd, int ifindex) {
         return setsockopt_int(fd, SOL_SOCKET, SO_BINDTOIFINDEX, ifindex);
 }
 
+int socket_autobind(int fd, char **ret_name) {
+        int r;
+
+        assert(fd >= 0);
+        assert(ret_name);
+
+        union sockaddr_union sa = {
+                .sa.sa_family = AF_UNIX,
+        };
+        if (bind(fd, &sa.sa, offsetof(union sockaddr_union, un.sun_path)) < 0)
+                return log_debug_errno(errno, "Failed to autobind AF_UNIX socket: %m");
+
+        r = getsockname_pretty(fd, ret_name);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to get autobind socket name: %m");
+
+        return 0;
```

Comment:
this is a nice helper.

but this raises the question in me if we maybe want to take this further if we have this helper: the kernel's autobind logic is kinda dumb because it uses a 32bit random value only, which according to the birthday paradox is a bit too likely to conflict. If we have common code anyway, maybe we want to just use a 64bit value, or maybe even 128bit value here (i.e. sd_id128 stuff?)